File: task3/src/main/java/com/kmnvxh222/task3/EditContactActivity.java
package com.kmnvxh222.task3;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.text.Editable;
import android.text.TextWatcher;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;

public class EditContactActivity extends AppCompatActivity {

    private String TAG = "EditContactActivity";

    private String id;
    private EditText editTextName;
    private EditText editTextInfo;
    private Contacts editСontact;
    private Button buttonRemove;
    private Toolbar toolbar;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_edit_contact);

        editTextName = findViewById(R.id.editTextName_e);
        editTextInfo = findViewById(R.id.editTextInfo_e);
        buttonRemove = findViewById(R.id.removeButton);
        toolbar = findViewById(R.id.toolbar_e);

        toolbar.setNavigationOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                saveEditData(editСontact);
            }
        });

        getData();
        removeData();

    }

    private void getData() {
        editСontact = (Contacts) getIntent().getSerializableExtra("EDIT_CONTACT");
        id = (String) getIntent().getSerializableExtra("ID");
        if(editСontact != null) {
            editTextName.setText(editСontact.getName());
            editTextInfo.setText(editСontact.getInfo());
        }
        editData(editСontact);
    }


    private void editData(final Contacts edit_contact) {

        editTextName.addTextChangedListener(new TextWatcher() {
            @Override
            public void beforeTextChanged(CharSequence s, int start, int count, int after) {
            }

            @Override
            public void onTextChanged(CharSequence s, int start, int before, int count) {
            }

            @Override
            public void afterTextChanged(Editable s) {
                edit_contact.setName(s.toString());
            }
        });

        editTextInfo.addTextChangedListener(new TextWatcher() {
            @Override
            public void beforeTextChanged(CharSequence s, int start, int count, int after) {
            }

            @Override
            public void onTextChanged(CharSequence s, int start, int before, int count) {
            }

            @Override
            public void afterTextChanged(Editable s) {
                edit_contact.setInfo(s.toString());
            }
        });
    }

    private void saveEditData(Contacts editContact) {
        if(editContact != null) {
            Intent intent = new Intent(EditContactActivity.this, MainActivity.class);
            intent.putExtra("EDITED_CONTACT", editContact);
            intent.putExtra("ID", editContact.getId());
            setResult(Activity.RESULT_OK, intent);
            finish();
        }
    }


    private void removeData() {
        buttonRemove.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                id = (String) getIntent().getSerializableExtra("ID");
                Intent intent = new Intent(EditContactActivity.this, MainActivity.class);
                intent.putExtra("REMOVE_CONTACT", id);
                setResult(Activity.RESULT_OK, intent);
                finish();
            }
        });
    }
}

File: task5/src/main/java/com/kmnvxh222/task5/EditContactActivity.kt
package com.kmnvxh222.task5

import android.database.sqlite.SQLiteDatabase
import android.os.Bundle
import android.text.Editable
import android.text.TextWatcher
import androidx.appcompat.app.AppCompatActivity
import com.kmnvxh222.task5.db.DBHelper
import kotlinx.android.synthetic.main.activity_edit_contact.editTextInfo
import kotlinx.android.synthetic.main.activity_edit_contact.editTextName
import kotlinx.android.synthetic.main.activity_edit_contact.removeButton
import kotlinx.android.synthetic.main.activity_edit_contact.toolbar

class EditContactActivity : AppCompatActivity() {

    private lateinit var db: SQLiteDatabase
    private lateinit var dbHelper: DBHelper

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_edit_contact)

        dataBaseInitialization()

        getData()

        toolbar.setNavigationOnClickListener { saveEditData(getData()) }
        removeButton.setOnClickListener { removeData(getData()?.id!!) }
    }

    private fun dataBaseInitialization() {
        dbHelper = DBHelper(this, null)
        db = dbHelper.writableDatabase
    }

    private fun getData(): Contacts? {
        val id = intent.getSerializableExtra("ID") as String
        val editContact = dbHelper.getContactByID(id, db)
        editTextName.setText(editContact?.name)
        editTextInfo.setText(editContact?.info)
        editData(editContact!!)
        return editContact
    }

    private fun editData(editContact: Contacts) {
        editTextName.addTextChangedListener(object : TextWatcher {
            override fun beforeTextChanged(s: CharSequence, start: Int, count: Int, after: Int) {}

            override fun onTextChanged(s: CharSequence, start: Int, before: Int, count: Int) {}

            override fun afterTextChanged(s: Editable) {
                editContact.name = s.toString()
            }
        })

        editTextInfo.addTextChangedListener(object : TextWatcher {
            override fun beforeTextChanged(s: CharSequence, start: Int, count: Int, after: Int) {}

            override fun onTextChanged(s: CharSequence, start: Int, before: Int, count: Int) {}

            override fun afterTextChanged(s: Editable) {
                editContact.info = s.toString()
            }
        })
    }

    private fun saveEditData(editContact: Contacts?) {
        if (editContact != null) {
            dbHelper.updateContact(editContact, db)
            finish()
        }
    }

    private fun removeData(id: String) {
        dbHelper.deleteContact(id, db)
        finish()
    }

    override fun onDestroy() {
        super.onDestroy()
        db.close()
    }
}